Content.Client/_DV/CosmicCult/Visuals/MonumentPlacementPreviewOverlay.cs

```diff
@@ -0,0 +1,150 @@
+using System.Numerics;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.Player;
+using Robust.Shared.Enums;
+using Robust.Shared.Map;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Timing;
+using Robust.Shared.Utility;
+
+namespace Content.Client._DV.CosmicCult.Visuals;
+
+public sealed class MonumentPlacementPreviewOverlay : Overlay
+{
+    private readonly IEntityManager _ent;
+    private readonly IPlayerManager _player;
+    private readonly SpriteSystem _sprite;
+    private readonly SharedMapSystem _map;
+    private readonly MonumentPlacementPreviewSystem _preview;
+    private readonly IGameTiming _timing;
+    public override OverlaySpace Space => OverlaySpace.WorldSpaceEntities;
+
+    private readonly ShaderInstance _saturationShader;
+    private readonly ShaderInstance _unshadedShader;
+    private readonly ShaderInstance _starsShader;
+
+    public bool LockPlacement = false;
+    private EntityCoordinates _lastPos = new();
+
+    //for a slight fade in / out
+    //ss14's formatting settings can take my needlessly public variables away from my cold, dead hands
+    public float fadeInProgress = 0;
+    public float fadeInTime = 0.25f;
+    public bool fadingIn = true;
+
+    public float fadeOutProgress = 0;
+    public float fadeOutTime = 0.25f;
+    public bool fadingOut = false;
+
+    public float alpha = 0;
+
+    private SpriteSpecifier mainTex;
+    private SpriteSpecifier outlineTex;
+    private SpriteSpecifier starTex;
```

Comment:
Fit the names to the other code
```suggestion
    //for a slight fade in / out
    public float FadeInProgress = 0;
    public float FadeInTime = 0.25f;
    public bool FadingIn = true;

    public float FadeOutProgress = 0;
    public float FadeOutTime = 0.25f;
    public bool FadingOut = false;

    public float Alpha = 0;

    private readonly SpriteSpecifier _mainTex;
    private readonly SpriteSpecifier _outlineTex;
    private readonly SpriteSpecifier _starTex;
```